presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java

```diff
@@ -154,6 +168,9 @@ public DispatchManager(
         this.queryTracker = new QueryTracker<>(queryManagerConfig, dispatchExecutor.getScheduledExecutor(), clusterQueryTrackerService);
 
         this.securityConfig = requireNonNull(securityConfig, "securityConfig is null");
+
+        this.queryMonitor = requireNonNull(queryMonitor, "queryMonitor is null");
+        this.queryHeartbeatExecutor = Executors.newSingleThreadScheduledExecutor(threadsNamed("query-heartbeat-management-%s"));
```

Comment:
Cleaner to instantiate this in CoordinatorModule and bind it to a annotated ScheduledExecutorService here - see CoordinatorModule.java for how we do this in other cases